### ISSUE TITLE
haraka crash when set nodemailer secure true

### ISSUE THREAD

**Issue by fflrichard** [baudehlo/Haraka#1183]
I'm studying haraka and first thank you for your excellent work.
I installed haraka in ubuntu and it works well with STARTTLS. I can send emails using the outlook client.
then I tried to use nodemailer, and something went wrong.

Haraka version 2.6.1 and nodemailer version 1.4.0
when i use following nodemailer options, haraka logs 'UNABLE_TO_GET_ISSUER_CERT', and i think it's nodemailer's problem.
```
var transporter = nodemailer.createTransport({
    host: '192.168.0.33',
    port: 2525,
    secure:false,
    auth:{
        user:'abc@abc.com',
        pass:'a112358'
    }
});
```

but when i set the secure option true, haraka crashed
```
var transporter = nodemailer.createTransport({
    host: '192.168.0.33',
    port: 2525,
    secure:true,
    auth:{
        user:'abc@abc.com',
        pass:'a112358'
    }
});
```

the output in node v0.10.35 is
```
[DEBUG] [25D38B82-6640-42F5-899B-3DE4040A7851] [core] running connect_init hooks
[DEBUG] [25D38B82-6640-42F5-899B-3DE4040A7851] [core] running connect_init_respond
[DEBUG] [25D38B82-6640-42F5-899B-3DE4040A7851] [core] running lookup_rdns hooks
[DEBUG] [25D38B82-6640-42F5-899B-3DE4040A7851] [core] [early_talker] state=4 esmtp=false line="C?���bU�Y	椸s�{�Ġb�<(������0�,�(�$��"
[CRIT] [-] [core] Caught exception: {}
[NOTICE] [-] [core] Shutting down
```
the output in node v4.1.0 is
```
[DEBUG] [AD2B5DE9-6EDA-4E0E-97D0-8BB9076CAB85] [core] running connect_init hooks
[DEBUG] [AD2B5DE9-6EDA-4E0E-97D0-8BB9076CAB85] [core] running connect_init_respond
[DEBUG] [AD2B5DE9-6EDA-4E0E-97D0-8BB9076CAB85] [core] running lookup_rdns hooks
[DEBUG] [AD2B5DE9-6EDA-4E0E-97D0-8BB9076CAB85] [core] [early_talker] state=4 esmtp=false line="�C �0�,�(�$��"
[CRIT] [-] [core] RangeError: Maximum call stack size exceeded
[CRIT] [-] [core]     at new Uint8Array (native)
[CRIT] [-] [core]     at Buffer.subarray (native)
[CRIT] [-] [core]     at Buffer.slice (buffer.js:565:23)
[CRIT] [-] [core]     at Connection._process_data (E:\workspace\node\Haraka\connection.js:357:43)
[CRIT] [-] [core]     at Connection._process_data (E:\workspace\node\Haraka\connection.js:374:18)
[CRIT] [-] [core]     at Connection._process_data (E:\workspace\node\Haraka\connection.js:374:18)
[CRIT] [-] [core]     at Connection._process_data (E:\workspace\node\Haraka\connection.js:374:18)
[CRIT] [-] [core]     at Connection._process_data (E:\workspace\node\Haraka\connection.js:374:18)
[CRIT] [-] [core]     at Connection._process_data (E:\workspace\node\Haraka\connection.js:374:18)
[CRIT] [-] [core]     at Connection._process_data (E:\workspace\node\Haraka\connection.js:374:18)
[NOTICE] [-] [core] Shutting down
```

of course nodemailer is not good enough, but i think there should no crash here.

**Comment by Dexus** [comment]
Hi, 

what do you have configure in config/tls.ini on haraka?

what is get nodemailer on callback for an error?

```
transporter.on("log", function(data) {console.log("LOG type: " + data.type);console.log("LOG message: " + data.message);});

// send mail
transporter.sendMail({
    from: 'sender@example.com',
    to: 'receiver@example.com',
    subject: 'hello world!',
    text: 'Authenticated with OAuth2'
},
function(err,data){
    console.log(err, data);
});
```

**Comment by smfreegard** [comment]
Ok - that bug is already fixed and once Haraka 2.7.0 is out, it will no longer crash.

However - you're incorrectly setting 'secure: true' in Nodemailer.   The secure flag puts Nodemailer into SMTPS mode, which is used on port 465 only (it's also deprecated in favour of port 587 as STARTTLS) and you're attempting that on port 25 (which is plain-text only).

So - in short - drop the `secure: true` from your Nodemailer options and it will work just fine and Nodemailer will use STARTTLS instead (provided you have configured Haraka with TLS certificates).

**Comment by fflrichard** [comment]
i know secure:true means ssl, my idea is just  we need no crash
